core/CMakeLists.txt

```diff
@@ -59,44 +59,36 @@ macro (nn_check_lib LIB SYM DEF)
 endmacro (nn_check_lib)
 
 set(gateway_c_sources
+    ${dynamic_library_c_file}
     ./src/message.c
     ./src/message_queue.c
-    ./src/internal/event_system.c
-    ./src/gateway_internal.c
-    ./src/gateway.c
-    ./src/gateway_createfromjson.c
-    ./src/broker.c
     ./src/module_loader.c
-    ${dynamic_library_c_file}
 )
 
-
 set(gateway_h_sources
     ./inc/message.h
-    ./inc/message_queue.h
-    ./inc/broker.h
     ./inc/module.h
     ./inc/module_access.h
+    ./inc/module_loader.h
+    ./inc/dynamic_library.h
+    ../deps/parson/parson.h
```

Comment:
The module host process needs some units from the gateway.dll, but absolutely may **not** use other units in gateway.dll - specifically broker.c.  I created a second library called "module_host" which the native_module_host module uses.  I'm willing to entertain better ways to handle this.